File: relationship-manager/common/src/main/java/com/bank/rm/common/dto/ApiResponse.java
package com.bank.rm.common.dto;

import java.time.Instant;

public record ApiResponse<T>(
    boolean success,
    T data,
    ErrorDetail error,
    Instant timestamp
) {
    public ApiResponse(boolean success, T data) {
        this(success, data, null, Instant.now());
    }

    public ApiResponse(boolean success, T data, ErrorDetail error) {
        this(success, data, error, Instant.now());
    }

    public static <T> ApiResponse<T> ok(T data) {
        return new ApiResponse<>(true, data);
    }

    public static <T> ApiResponse<T> error(String code, String message) {
        return new ApiResponse<>(false, null, new ErrorDetail(code, message, null, null));
    }
}

File: relationship-manager/common/src/main/kotlin/com/bank/rm/common/dto/ApiResponse.kt
package com.bank.rm.common.dto

import java.time.Instant

data class ApiResponse<T>(
    val success: Boolean,
    val data: T? = null,
    val error: ErrorDetail? = null,
    val timestamp: Instant = Instant.now()
)

data class ErrorDetail(
    val code: String,
    val message: String,
    val details: List<FieldError>? = null,
    val traceId: String? = null
)

data class FieldError(
    val field: String,
    val message: String,
    val rejectedValue: Any? = null
)

data class PagedResponse<T>(
    val content: List<T>,
    val page: Int,
    val size: Int,
    val totalElements: Long,
    val totalPages: Int
)
